core/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -112,6 +111,16 @@
         Setting.byteSizeSetting("index.translog.flush_threshold_size", new ByteSizeValue(512, ByteSizeUnit.MB), Property.Dynamic,
             Property.IndexScope);
 
+    /**
+     * The maximum size of a translog generation. This is independent of the maximum size of
+     * translog operations that have not been flushed.
+     */
+    public static final Setting<ByteSizeValue> INDEX_TRANSLOG_GENERATION_THRESHOLD_SIZE_SETTING =
+            Setting.byteSizeSetting(
+                    "index.translog.generation_threshold_size",
```

Comment:
I picked `index.translog.generation_threshold_size` to be consistent with `index.translog.flush_threshold_size`. Do you still wonder if it should be changed?